docs/setting-up-github-actions.md

```diff
@@ -0,0 +1,87 @@
+# Setting up SBOM generation for GitHub Actions
+
+This document provides an example on how to integrate SBOM tool into GitHub Actions, you may use this as a guide to adding the tool to your GitHub action.
+
+## Existing setup
+
+In our Github project, the source contains a project called Sample. We also have a workflow that builds the project and saves the generated binaries as a pipeline artifact.
+
+```yaml
+name: Sample
+
+on: 
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+
+    - name: Build
+      run: dotnet build Sample.sln --output buildOutput
+
+    - name: Upload a Build Artifact
+      uses: actions/upload-artifact@v3.1.0
+      with:
+        path: buildOutput
+```
+
+Upon generation, we see that the artifacts are uploaded to the Actions run page, the generated binaries and other files are placed in the artifact.
+
+![actions run](./images/github-workflow-run-details.png)
+![actions-artifact-without-sbom](./images/github-downloaded-folder-without-sbom.png)
+
+## Adding the SBOM generation task
+
+We will generate the SBOM for the build artifacts we generate in the previous step. We will store the generated SBOM as part of the build artifacts, as we will be distributing this artifact to our downstream dependencies. 
+
+```yaml
+name: Sample with SBOM generation
+
+on: 
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+    - name: Build
+      run: dotnet build Sample.sln --output buildOutput
+      
+    - name: Generate SBOM
+      run: |
+        curl -Lo sbom-tool https://github.com/microsoft/sbom-tool/releases/latest/download/sbom-tool-linux-x64
```

Comment:
This example will download sbom-tool to the same folder as repository which in some cases will add sbom-tool itself to files section in the output SBOM. It's not common scenario, but I could see it possible for some JS/TS projects where BuildDropPath is the same as the source code. Should we update that to download the tool to some isolated folder outside working directory to make sure people never have this scenario by using our example?

Reply:
Good idea, let me update that

Reply:
fixed